Code/Tools/FBuild/FBuildCore/Graph/FileNode.cpp

```diff
@@ -66,6 +69,205 @@ void FileNode::HandleWarningsClangCl( Job * job, const AString & name, const ASt
+                        {
+                            numberOfLastInclude = numberOfLastInclude * 10 + (*cursor - '0');
+                            ++cursor;
+                        }
+
+                        ++cursor; // skip ' '
+                        ASSERT(*cursor == '"');
+                        ++cursor; // skip '"'
+
+                        const char* filenameBegin = cursor;
+                        while (*cursor != '"')
+                        {
+                            ++cursor;
+                        }
+                        const char* filenameEnd = cursor;
+
+                        filename.Assign(filenameBegin, filenameEnd);
+                    }
+                }
+
+				startOfLine = false;
+            }
+
+			++cursor;
+        }
+
+        sourceLoc.lineNum = sourceLoc.lineNum - (lastIncludeLine + 1) + numberOfLastInclude;
+        PathUtils::FixupFilePath( filename );
+        sourceLoc.filename = filename;
+    }
+
+    return;
+}
+
+// HandleDiagnosticsClangTidy
+//------------------------------------------------------------------------------
+void FileNode::HandleDiagnosticsClangTidy( Job* job, const AString& name, const AString& data )
+{
+    if ( data.IsEmpty() )
+    {
+        return;
+    }
+
+    AString newWarningMessage;
+    // Avoid the client who wants to print remote work to process again the error line number
+    // This part of the function workaround this issue : https://github.com/llvm/llvm-project/issues/62405
+    if ( ( job->GetDistributionState() == Job::DIST_NONE || job->GetDistributionState() == Job::DIST_BUILDING_LOCALLY ) )
+    {
+
+        Array< AString > lines;
+        data.Tokenize( lines, '\n' );
+
+        Array< SourceLocation > sourceLocs;
+        Array< uint32_t > fixupLineIndices;
+        for (uint32_t i = 0; i < lines.GetSize(); ++i)
+        {
+            // Parse the warning message to get the error line numbers in preprocessed file
+            // The warning message looks like this : "path\to\the\file:errorLineNumber:errorColumnNumber: warning: description of what is wrong"
+            // There may be several concatenated diagnostic messages
+            // It works both on Windows and Unix systems
+
+            AString& line = lines[i];
+
+            // The error line number is just before the "warning" string which is not at the beginning
+            const char* diagnosticBegin = nullptr;
+            diagnosticBegin = line.Find(": error:");
+            diagnosticBegin = diagnosticBegin ? diagnosticBegin : line.Find(": warning:");
+            diagnosticBegin = diagnosticBegin ? diagnosticBegin : line.Find(": note:");
+
+            if (diagnosticBegin != nullptr)
+            {
+                fixupLineIndices.Append(i);
+
+                // Skip column number
+                const char* cursor = diagnosticBegin - 1; // skip starting ':'
+                while (*cursor >= '0' && *cursor <= '9')
+                {
+                    --cursor;
```

Comment:
I'm not sure if this is possible, but if the first list began with ": error:" (or similar) this could read before the start of the buffer.